### PR TITLE
[Tidy] Replace underscores with dashes in classNames

### LINKED ISSUE #722: Replace underscores with dashes in CSS classnames
- [ ] Do a search and find (e.g. `className=`) and replace all underscores with dashes in CSS classnames (except for vizro_dark and vizro_light)
- [ ] Replace double underscores with dashes (e.g. container__title to container-title)
- [ ] Ensure all tests pass

### PR DESCRIPTION
**Pull request by rilieo**
## Description

Fix #722
Replaced underscores with dashes in all occurrences of `className=` (except for `className="vizro_dark"`)
Changed respective class names in css files

## Screenshot

N/A

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Comment by huong-li-nguyen** [comment]
Hey @rilieo - thanks for taking this one! It probably wasn't clear from the ticket, but zou would have to change the equivalent in the CSS files as well, otherwise, the components will not be properly styled :)

So if you change `className=container__title` to `className=container-title`, 

you also have to change the CSS file respectively from

```
.container__title {
...
}

```
to 

```
.container-title {
..
}
```


A quick search and find in the CSS files should help 👍

File: vizro-core/src/vizro/models/_components/form/_alert.py
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/models/_components/form/_alert.py:45]: I just saw that this CSS class is not used anywhere in the CSS files, so feel free to remove it.

```suggestion
```

File: vizro-core/src/vizro/models/_components/form/date_picker.py
Comment (huong-li-nguyen) [inline vizro-core/src/vizro/models/_components/form/date_picker.py:111]: Doesn't seem to be used anywhere in the CSS files - feel free to remove.

```suggestion
```

File: vizro-core/tests/unit/vizro/models/_components/form/test_date_picker.py
Comment (huong-li-nguyen) [inline vizro-core/tests/unit/vizro/models/_components/form/test_date_picker.py:140]: Doesn't seem to be used anywhere in the CSS files - feel free to remove.

```suggestion
```

**Comment by huong-li-nguyen** [comment]
This looks good to go for me - thanks for contributing! 👏 

I left a couple of code suggestions to clean up the CSS (some of the classNames were not used in the CSS files). Other than that:

- [ ] Raise this from draft to PR
- [ ] One of the tests is failing (Checks for Vizro) - hopefully the error message is clear enough :) But essentially you need to add a changelog file. Take a look at the error message and follow the instructions. If not clear, let me know! 
- [ ] Could you rename your PR to `[Tidy] Replace underscores with dashes in classNames`


After that I can approve 👍

**Review by huong-li-nguyen — COMMENTED** [review]
Congratulations on your first PR in this repository! 👏

**Review by huong-li-nguyen — APPROVED** [review]
LGTM 👍 🚀  You might have to wait until next week for this to get merged, as you need two approvals. The majority of our developers are back on Monday, but we'll make sure this gets merged 👍  Thank you so much for contributing!

**Review by lingyielia — APPROVED** [review]
Thanks for contributing! 💯

**Review by nadijagraca — APPROVED** [review]
Thank you very much for your contribution! 🙌  

You can add your name in docs/pages/explanation/authors.md, under the the ## Previous team members and code contributors section as you are the Vizro contributor now.  🎉 

Once you merge latest changes from main to this branch, we can merge the PR.

**Pull request merged**
